src/client/pps/util.go

```diff
@@ -129,3 +130,61 @@ func ValidateGitCloneURL(url string) error {
 
 	return nil
 }
+
+// JobStateFromName attempts to interpret a string as a JobState,
+// accepting either the enum names or the pretty printed state names
+func JobStateFromName(name string) (JobState, error) {
+	var state JobState
+	if value, ok := JobState_value[name]; ok {
+		state = JobState(value)
+	} else {
+		// check for pretty-printed name
+		switch strings.ToLower(name) {
```

Comment:
Very minor thought: would it make sense to implement the pretty-printed branch with something like:
```
...
} else if value, ok := JobState_value["JOB_"+strings.ToUpper(name)]; ok {
  state=JobState(value)
}
```
it's a little more mental gymnastics, but then we wouldn't have to worry about updating the switch statement if states are added or removed (we're talking about changing how pipeline states work soon, actually), and I think has the same result.

Reply:
Second thought: You might be able to compress both branches into a single statement:
```
var errInvalueJobState error
func init() {
  var msg bytes.Buffer
  for i int64 := 0; ; i++ {
    if name, ok := JobState_name[i]; ok {
      msg.WriteString(
  }
}

if value, ok := JobState_value["JOB_" + strings.TrimPrefix("JOB_", strings.ToUpper(name))]; ok {
  return JobState(value), nil
}
return nil,

Reply:
done, I guess that means neither of you had any comments about the error/help strings?

Reply:
error/help strings LGTM